File: 프로그래머스/lv2/42578. 의상/의상.java
import java.util.*;
class Solution {
    public int solution(String[][] clothes) {
        int answer = 1;
        HashMap<String, List<String>> map = new HashMap<>();
        
        for (int i = 0; i < clothes.length; i++) {
            List<String> keyCheck = map.get(clothes[i][1]);
            if (keyCheck == null) {
                keyCheck = new ArrayList<>();
                map.put(clothes[i][1], keyCheck);
            }
            keyCheck.add(clothes[i][0]);
        }

        for (String key : map.keySet()) {
            answer *= map.get(key).size() + 1;
        }
        return answer - 1;
    }
}

File: 프로그래머스/2/42578. 의상/의상.py
def solution(clothes):
    # 종류를 key, 그 수를 dict로 저장. 
    # item()로 순환하면서 각 종류를 곱한 값 - 1(아예 안입은 경우를 제외)
    answer = 1
    dict = {}
    for name, type_name in clothes:
        dict[type_name] = dict.get(type_name, 0) + 1
    
    for k, v in dict.items():
        answer *= v + 1
    
    return answer - 1
    
    